### PR TITLE
Updated Allatori Transformers + Stringer Fixes

### PR DESCRIPTION
**Pull request by ThisTestUser**
I added an Allatori flow deobfuscator. Since Allatori's flow obfuscation got improved, the transformer will greatly improve the readability of the code.
The Stringer "fix" uses brute force to determine where the "new" opcode should be placed in constructor obfuscation.
FILES CHANGED (OTHER FILES MERGE IS NOT NEEDED):
ArgsAnalyzer.java - Update
JVMMethodProvider.java - Small bug fixed + more methods
allatori/StringEncryptionTransformer.java (OLD) - Fix detector
allatori/FlowObfuscationTransformer.java - Remove allatori flow obfuscator
allatori/LightFlowObfuscationTransformer - Remove allatori dead code
stringer/HideAccessObfuscationTransformer - "Fallback" method to place the new opcode + fixes
smoke/NumberObfuscationTransformer - Better int handling
peephole/ConstantFolder - Fix
Utils.java - Some fixes
JavaMethod.class - Fix getMethods() and getFields()

### CODE REVIEW

**Comment by samczsun** [comment]
I'll take your word for it. However, it's hard to understand what sorts of transformations the code tries to remove. Could you give some examples in this PR?

File: src/main/java/com/javadeobfuscator/deobfuscator/utils/Utils.java [hunk 1]
```diff
@@ -376,8 +369,10 @@ public static int getIntValue(AbstractInsnNode node)
 		if(node.getOpcode() >= Opcodes.ICONST_M1
 			&& node.getOpcode() <= Opcodes.ICONST_5)
 			return node.getOpcode() - 3;
-		if(node.getOpcode() == Opcodes.SIPUSH
-			|| node.getOpcode() == Opcodes.BIPUSH)
+		if (node.getOpcode() == Opcodes.BIPUSH) {
```
[the remark this answers is not in the document]
Comment (ThisTestUser) [reply src/main/java/com/javadeobfuscator/deobfuscator/utils/Utils.java]: Having that doesn't seem to work (I don't remember where though)

**Comment by netherfoam** [comment]
Allatori flow deobfuscation? Sounds like exactly what I'm looking for!

**Comment by ThisTestUser** [comment]
PR too complex, closing.

**Pull request closed without merging**